lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -2943,8 +2943,10 @@ transactionFee DBLayer{atomically, walletsDB} protocolParams txLayer
                     $ ExceptionNoSuchWallet
                     $ ErrNoSuchWallet walletId
                 Just ws -> pure $ WalletState.getLatest ws
-        let utxoIndex = UTxOIndex.fromMap . CS.toInternalUTxOMap $
-                availableUTxO @s mempty wallet
+        utxoIndex <- evaluate
+            $ UTxOIndex.fromMap
+            $ CS.toInternalUTxOMap
+            $ availableUTxO @s mempty wallet
```

Comment:
As an experiment, I tried changing this to `evaluateDeep`, but saw no significant change in performance. This is a reasonable indication that evaluating to WHNF is enough to ensure that the `UTxOIndex` is fully evaluated.

Reply:
Just curious, could `$!` etc also have worked, or possibly `seq`?

Reply:
> Just curious, could `$!` etc also have worked, or possibly `seq`?

I think it will work, assuming you mean something like `seq utxoIndex (calculateFeePercentiles ...)`.

However, I think it's slightly nicer to keep the evaluation to WHNF close to the definition of `utxoIndex`. What do you think?